WooCommerce/Classes/ViewRelated/Orders/Order Creation/Synchronizer/OrderTotalsCalculator.swift

```diff
@@ -0,0 +1,62 @@
+import Foundation
+import Yosemite
+
+/// Helper to calculate the totals on an `Order`.
+///
+final class OrderTotalsCalculator {
+    // MARK: Private properties
+
+    private let order: Order
+
+    private let currencyFormatter: CurrencyFormatter
+
+    /// Total shipping amount on an order.
+    ///
+    private var shippingTotal: NSDecimalNumber {
```

Comment:
Any reason on why using `NSDecimalNumber` instead of `Decimal`?

Reply:
Mostly because I didn't see a compelling reason to change it from what we were already using. Most of the calculations are extracted directly from `NewOrderViewModel.configurePaymentDataViewModel()`, and the `convertToDecimal` method we've been using from `CurrencyFormatter` returns `NSDecimalNumber`.